src/lib/y2network/dialogs/s390_device_activation.rb

```diff
@@ -75,23 +80,26 @@ def contents
       end
 
       def run
-        ret = super
-        if ret == :next
-          configured = activator.configure
-          builder.name = activator.configured_interface if configured
-          # TODO: Refresh the list of interfaces in yast_config. Take into
-          # account that the interface in yast_config does not have a name so
-          # the builder.interface is probably nil and should be obtained
-          # through the busid.
-          if !configured || builder.name.empty?
-            Yast::Popup.Error(
-              _(
-                "An error occurred while creating device.\nSee YaST log for details."
-              )
-            )
-
-            ret = nil
+        ret = nil
+        loop do
+          ret = super
+          if ret == :next
+            _stdout, stderr, status = activator.configure
+            configured = status.zero?
+
+            if configured
+              interface_name = activator.configured_interface
+              builder.name = interface_name
+              add_interface(interface_name)
+            end
+
+            if !configured || builder.name.empty?
+              show_activation_error(stderr)
+              ret = :redraw
```

Comment:
@jreidinger that is what you suggest, isn't it? apart of removing the loop of course

```suggestion
              return run
```